## Extraction du text

In [32]:
import pdfplumber
import pandas as pd
import google.generativeai as genai
import pypdf
import google.generativeai as genai
import time


In [16]:
pdf_path = "docu.pdf"

In [17]:

import pypdf

def extract_text_from_pdf(pdf_path):
    with open(pdf_path, "rb") as f:
        reader = pypdf.PdfReader(f)
        text = "\n".join(page.extract_text() or "" for page in reader.pages)
    return text




text = extract_text_from_pdf(pdf_path)




Identifier les sections importantes du texte

In [25]:
import re

# Mots-clés pour chaque KPI
kpi_keywords = {
    "Gouvernance": ["Conseil d'Administration", "Conseil de Surveillance", "Comité"],
    "Rémunération Fixe": ["rémunération fixe", "salaire de base", "indemnité"],
    "Rémunération Variable": ["bonus", "prime", "performance"],
    "Fréquence Réunions": ["réunion", "fréquence", "nombre de séances"],
    "Rémunération par Responsabilité": ["montant", "rémunération selon poste"],
    "Chiffres Clés": ["CA", "dette", "bénéfice", "résultat net", "capitalisation"]
}



# Extraction des passages contenant ces mots-clés
def extract_kpi_data(text, kpi_keywords):
    kpi_data = {key: [] for key in kpi_keywords}  # Initialisation
    for kpi, keywords in kpi_keywords.items():
        for keyword in keywords:
            matches = re.findall(rf".*{keyword}.*", text, re.IGNORECASE)
            if matches:
                kpi_data[kpi].extend(matches)
    return kpi_data


extracted_data = extract_kpi_data(text, kpi_keywords)


In [ ]:
# Configurez l'API avec votre clé
genai.configure(api_key="AIzaSyDcoBS-YeFZoDKrg4thhz5M1dNfcDTuuuU")

model = genai.GenerativeModel("gemini-pro")

summary = {}

for kpi, content in extracted_data.items():
    if content:
        prompt = f"Voici des informations extraites du document sur {kpi} :\n\n{'\n'.join(content)}\n\nPeux-tu résumer ces informations de manière concise ?"
        try:
            response = model.generate_content(prompt)
            summary[kpi] = response.text
            time.sleep(2)  # Pause pour éviter les erreurs de quota
        except Exception as e:
            print(f"Erreur avec {kpi}: {e}")

    time.sleep(5)
    
# Affichage du résumé
for kpi, res in summary.items():
    print(f"\n--- {kpi} ---\n{res}")


Erreur avec Chiffres Clés: 429 Resource has been exhausted (e.g. check quota).

--- Gouvernance ---
**Gouvernance**

**Conseil d'administration**

* 22 membres, dont 21 représentants de la direction et du CA
* Veille à l'adéquation des compétences et fonctions des membres du CA

**Comités du CA**

**Comité d'audit et des risques**
* Assure la surveillance des contrôles internes et des risques
* Valide le plan d'audit et le rapport des commissaires aux comptes
* Reçoit les alertes et en fait rapport au CA

**Comité des rémunérations, des nominations et de la RSE**
* Responsable des rémunérations, des nominations et de la politique RSE
* Évalue la performance du CA et des comités
* Fait des recommandations au CA sur les questions de RSE

**Comité des engagements**
* Prépare les décisions du CA sur les engagements financiers
* Émet des avis sur les opérations stratégiques
* Valide les plans de succession de la direction

**Comité exécutif**

* Dirigé par le Président-directeur général
* R

In [29]:
summary

{'Gouvernance': "**Gouvernance d'entreprise**\n\n* Membres du conseil d'administration et du comité exécutif (hors charges) : 22\n* Nombre de comités au sein du conseil d'administration : 3\n* Président du comité d'audit et des risques : Florent André\n* Président du comité des rémunérations, nominations et RSE : Yann Demolière\n* Président du comité des engagements : Bruno Deserti",
 'Rémunération Fixe': "**Clé** | **Valeur**\n------- | --------\nRémunération fixe brute annuelle | 1 030 000\xa0€\nRémunération fixe brute annuelle 2023 | 1 030 000\xa0€\nPart variable annuelle | 0% à 120% de la rémunération fixe\nPart variable annuelle 2023 | 105% de la rémunération fixe\nIndemnité de cessation de fonction | Égale à la rémunération annuelle brute (cible atteinte)\nIndemnité de cessation de fonction en cas de non-renouvellement à l'initiative du PDG | 75% de la rémunération annuelle brute (cible atteinte)\nIndemnité de cessation de fonction en cas de non-renouvellement hors de l'initiativ

# Essayer sorti JSON:


{'Gouvernance': "**Gouvernance d'entreprise**\n\n* Membres du conseil d'administration et du comité exécutif (hors charges) : 22\n* Nombre de comités au sein du conseil d'administration : 3\n* Président du comité d'audit et des risques : Florent André\n* Président du comité des rémunérations, nominations et RSE : Yann Demolière\n* Président du comité des engagements : Bruno Deserti",
 'Rémunération Fixe': "**Clé** | **Valeur**\n------- | --------\nRémunération fixe brute annuelle | 1 030 000\xa0€\nRémunération fixe brute annuelle 2023 | 1 030 000\xa0€\nPart variable annuelle | 0% à 120% de la rémunération fixe\nPart variable annuelle 2023 | 105% de la rémunération fixe\nIndemnité de cessation de fonction | Égale à la rémunération annuelle brute (cible atteinte)\nIndemnité de cessation de fonction en cas de non-renouvellement à l'initiative du PDG | 75% de la rémunération annuelle brute (cible atteinte)\nIndemnité de cessation de fonction en cas de non-renouvellement hors de l'initiative du PDG | 100% de la rémunération annuelle brute (cible atteinte)",
 'Rémunération Variable': "**Clé**|**Valeur**\n-------|-------\nPrimes|Liées à la performance, non liées à la rémunération annuelle brute\nPrimes d'émission|Amortissement accéléré\nPrimes de fidélité|Comptabilisées en résultat financier lors de leur survenance\nPrise en charge des frais|Destinée aux employés\nIncorporation de réserves et bénéfices|Peut être décidée par l'entité\nPerformance financière et extra-financière|Inclut les indicateurs clés de performance et les informations prévues par la réglementation\nPerformance durable|Objectifs liés à l'attribution gratuite d'actions de performance\nPerformance RSE|Mesurée selon les normes de la Global Reporting Initiative (GRI)\nConditions de performance|Mesurées entre le 1er janvier 2023 et la fin de la période de conservation\nPlans d'actions sous conditions de performance|Attribués par le groupe Edenred\nPerformance économique|Inclus dans l'étude sur la performance économique et fiscale du secteur\nPerformance ESG|Mesurée et évaluée par Edenred\nEngagement des collaborateurs|Suivi à l'échelle du groupe\nEngagement en matière de données|Mesuré via des indicateurs clés de performance\nEngagement d'accompagnement|Mesuré par des indicateurs clés de performance\nindicateurs clés de performance |Utilisés pour évaluer la performance et sont validés par le Comité exécutif du groupe",
 'Fréquence Réunions': "**Clé** | **Valeur**\n---|---|\n**Réunions actionnaires** | 8, 4, 4, 5\n**Réunions du Comité des rémunérations** | 6\n**Réunions du Conseil d'administration** | 12\n**Réunions exceptionnelles du Conseil d'administration** | 3\n**Réunion des administrateurs indépendants** | 1\n**Réunions du Comité d'audit et des risques** | 3\n**Réunions consacrées aux comptes semestriels et annuels** | 2\n**Réunions du Comité des engagements** | 5\n**Réunions du Comité Diversité** | 2\n**Durée moyenne des réunions du Conseil d'administration** | 4h15\n**Durée moyenne des réunions du Comité d'audit et des risques** | 2h\n**Durée moyenne des réunions du Comité des engagements** | 1h20",
 'Rémunération par Responsabilité': "- Montant de l'amende infligée à Edenred : 158 millions d'euros\n- Solde de trésorerie d'Edenred fin 2023 : 158 millions d'euros\n- Capital social et fonds propres d'Edenred : montant non indiqué\n- Montant des prêts interentreprises : montant non indiqué\n- Montant des dividendes versés en 2022 : 1,00 € par action\n- Montant de la provision pour risques au 31 décembre 2023 : 249 millions d'euros\n- Montant de la charge financière en 2023 : 218 millions d'euros\n- Montant des intérêts versés en 2023 : 98 millions d'euros\n- Montant de l'augmentation de capital réalisée en 2023 : 1,2 milliard d'euros\n- Montant de la rémunération du président-directeur général : montant non indiqué"}